graylog2-web-interface/src/integrations/aws/cloudtrail/StepAuthorize.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import React, { useEffect, useContext, useState } from 'react';
+
+import FormDataContext from 'integrations/contexts/FormDataContext';
+import { qualifyUrl } from 'util/URLUtils';
+import fetch from 'logic/rest/FetchProvider';
+import formValidation from 'integrations/aws/utils/formValidation';
+import { AWS_AUTH_TYPES } from 'integrations/aws/common/constants';
+
+import { ApiRoutes } from './common/Routes';
+import type { ErrorMessageType, HandleFieldUpdateType, HandleSubmitType } from './types';
+
+import { renderOptions } from '../common/Options';
+import FormWrap from '../common/FormWrap';
+import ValidatedInput from '../common/ValidatedInput';
+import AWSAuthenticationTypes from '../authentication/AWSAuthenticationTypes';
+
+
+type StepAuthorizeProps = {
+  onSubmit: HandleSubmitType;
+  onChange: HandleFieldUpdateType;
+};
+
+const StepAuthorize = ({ onSubmit, onChange }: StepAuthorizeProps) => {
+  const [regions, setRegions] = useState([]);
+
+  useEffect(() => {
+    fetch('GET', qualifyUrl(ApiRoutes.INTEGRATIONS.AWSCloudTrail.GET_AWS_REGIONS)).then((response) => {
+      const region = Object.keys(response).map((key) => ({ label: response[key], value: key }));
+      setRegions(region);
+    });
+  }, []);
+
+  const { formData } = useContext(FormDataContext);
+  const [formError, setFormError] = useState<ErrorMessageType>(null);
+
+  const { awsSecretKey } = formData;
+  const [loading, setLoading] = useState(false);
+
+  const handleSubmit = () => {
+    setLoading(true);
+
+    fetch('POST', qualifyUrl(ApiRoutes.INTEGRATIONS.AWSCloudTrail.CHECK_CREDENTIALS), {
+      aws_access_key: formData?.awsAccessKey?.value || '',
+      aws_secret_key: formData?.awsSecretKey?.value || '',
+      cloudtrail_queue_name: formData?.awsCloudTrailSqsQueueName?.value || '',
+      aws_region: formData?.awsCloudTrailRegion?.value || '',
+      assume_role_arn: formData?.awsAssumeRoleARN?.value || '',
```

Comment:
Not sure if it matters but we use `awsAssumeRoleArn` everywhere else.
```suggestion
      assume_role_arn: formData?.awsAssumeRoleArn?.value || '',
```

Reply:
> Not sure if it matters but we use `awsAssumeRoleArn` everywhere else.

awsAssumeRoleARN this is used in whole project.Do you want me to update this?